File: midonet-cluster/src/main/java/org/midonet/cluster/rest_api/ResourceUris.java
/*
 * Copyright 2015 Midokura SARL
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.midonet.cluster.rest_api;

import java.util.UUID;

import org.midonet.packets.MAC;

/**
 * This class contains the name of all the resources that are exposed via our
 * REST API.
 */
public final class ResourceUris {
    public static final String ARP_TABLE = "arp_table";
    public static final String BGP_NETWORKS = "bgp_networks";
    public static final String BGP_PEERS = "bgp_peers";
    public static final String BINDINGS = "bindings";
    public static final String BRIDGES = "bridges";
    public static final String CHAINS = "chains";
    public static final String DHCP = "dhcp";
    public static final String DHCPV6 = "dhcpV6";
    public static final String DHCPV6_HOSTS = "hostsV6";
    public static final String DHCP_HOSTS = "hosts";
    public static final String HEALTH_MONITORS = "health_monitors";
    public static final String HOSTS = "hosts";
    public static final String INTERFACES = "interfaces";
    public static final String IP_ADDRS = "ip_addrs";
    public static final String IP_ADDR_GROUPS = "ip_addr_groups";
    public static final String LINK = "link";
    public static final String LOAD_BALANCERS = "load_balancers";
    public static final String L2INSERTIONS = "l2insertions";
    public static final String L2SERVICES = "l2services";
    public static final String MAC_TABLE = "mac_table";
    public static final String MIRRORS = "mirrors";
    public static final String PEER_PORTS = "peer_ports";
    public static final String POOLS = "pools";
    public static final String POOL_MEMBERS = "pool_members";
    public static final String PORTS = "ports";
    public static final String PORT_GROUPS = "port_groups";
    public static final String ROUTERS = "routers";
    public static final String ROUTES = "routes";
    public static final String RULES = "rules";
    public static final String SYSTEM_STATE = "system_state";
    public static final String TENANTS = "tenants";
    public static final String TRACE_REQUESTS = "traces";
    public static final String TUNNEL_ZONES = "tunnel_zones";
    public static final String VERSIONS = "versions";
    public static final String VIPS = "vips";
    public static final String VLANS = "vlans";
    public static final String VTEPS = "vteps";
    public static final String VTEP_BINDINGS = "vtep_bindings";
    public static final String VXLAN_PORTS = "vxlan_ports";
    public static final String WRITE_VERSION = "write_version";

    // TODO: remove these? they are templates, and belong in the resources IMO
    public static final String TENANT_ID_PARAM = "tenant_id";

    public static String macToUri(String mac) {
        return mac.replace(':', '-');
    }

    public static MAC macPortUriToMac(String macPortAsUri) {
        return MAC.fromString(macPortAsUri.split("_")[0].replace('-', ':'));
    }

    public static UUID macPortUriToPort(String macPortAsUri) {
        return UUID.fromString(macPortAsUri.split("_")[1]);
    }

}

File: midonet-cluster/src/main/java/org/midonet/cluster/rest_api/ResourceUris.scala
/*
 * Copyright 2015 Midokura SARL
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.midonet.cluster.rest_api

/**
 * This class contains the name of all the resources that are exposed via our
 * REST API.
 */
object ResourceUris {
    val ARP_TABLE: String = "arp_table"
    val BGP_NETWORKS: String = "bgp_networks"
    val BGP_PEERS: String = "bgp_peers"
    val BINDINGS: String = "bindings"
    val BRIDGES: String = "bridges"
    val CHAINS: String = "chains"
    val DHCP: String = "dhcp"
    val DHCPV6: String = "dhcpV6"
    val DHCPV6_HOSTS: String = "hostsV6"
    val DHCP_HOSTS: String = "hosts"
    val HEALTH_MONITORS: String = "health_monitors"
    val HOSTS: String = "hosts"
    val INTERFACES: String = "interfaces"
    val IP_ADDRS: String = "ip_addrs"
    val IP_ADDR_GROUPS: String = "ip_addr_groups"
    val LINK: String = "link"
    val LOAD_BALANCERS: String = "load_balancers"
    val L2INSERTIONS: String = "l2insertions"
    val MAC_TABLE: String = "mac_table"
    val MIRRORS: String = "mirrors"
    val PEER_PORTS: String = "peer_ports"
    val PEERING_TABLE: String = "peering_table"
    val POOLS: String = "pools"
    val POOL_MEMBERS: String = "pool_members"
    val PORTS: String = "ports"
    val PORT_GROUPS: String = "port_groups"
    val ROUTERS: String = "routers"
    val ROUTES: String = "routes"
    val RULES: String = "rules"
    val SYSTEM_STATE: String = "system_state"
    val SERVICE_CONTAINERS: String = "service_containers"
    val SERVICE_CONTAINER_GROUPS: String = "service_container_groups"
    val TENANTS: String = "tenants"
    val TRACE_REQUESTS: String = "traces"
    val TUNNEL_ZONES: String = "tunnel_zones"
    val VERSIONS: String = "versions"
    val VIPS: String = "vips"
    val VLANS: String = "vlans"
    val VTEPS: String = "vteps"
    val VTEP_BINDINGS: String = "vtep_bindings"
    val VXLAN_PORTS: String = "vxlan_ports"
    val TENANT_ID_PARAM: String = "tenant_id"

    def macToUri(mac: String): String = {
        mac.replace(':', '-')
    }

}
